rotkehlchen/exchanges/bitpanda.py

```diff
@@ -208,18 +210,30 @@ def _deserialize_wallettx(
             )
             return None
 
-        return AssetMovement(
-            location=Location.BITPANDA,
-            category=movement_category,
-            address=address,
-            transaction_id=transaction_id,
-            timestamp=time,
+        events = [AssetMovement(
+            location=self.location,
+            event_type=event_type,
+            timestamp=(timestamp := ts_sec_to_ms(time)),
             asset=asset,
-            amount=amount,
-            fee_asset=asset,
-            fee=fee,
-            link=tx_id,
-        )
+            balance=Balance(abs(amount)),
+            unique_id=f'{tx_id}{transaction_id}',  # Use both here as tx_id is not always unique (at least in the test data)  # noqa: E501
+            extra_data=maybe_set_transaction_extra_data(
+                address=address,
+                transaction_id=transaction_id,
+            ),
+        )]
+        if fee != ZERO:
```

Comment:
same as in bitfinex and here this would be simplified and reduce code with the abstracted function